### ISSUE TITLE
Is there a way to check when loading a game save fails?

### ISSUE THREAD

**Issue by TheSkyOne** [Iakobs/godot-play-game-services#38]
i have connected the `game_loaded` signal to a function that will take the snapshot and process its info.
but it seems that when there is no save to be loaded, that function i connected simply doesnt get run (i presume the game_loaded signal isnt being fired?)
is there a way for me to run some code for when there is no save to be loaded?

**Comment by Iakobs** [comment]
Hi! Yes, right now the signal is not triggered if there's no game to be loaded. I could just return send the signal with a null object in that case, I think it makes sense in case you want to react to that scenario. Let me check if I can implement this change and I will let you know!

**Comment by Iakobs** [comment]
Hi! Try the [new release](https://github.com/Iakobs/godot-play-game-services/releases/tag/v1.8.3), it's now sending a null in the `game_loaded` signal. Remember also that you can set to `true` the second parameter of the `load_game` method when loading a game. That will create the Snapshot if it doesn't exist.

**Comment by TheSkyOne** [comment]
so to be clear on the usage, in my function that is connected to `game_loaded` i can now check
```
if snapshot:
    #code when snapshot is found
else:
    #code when no snapshot was found
```
also, for installing this new release, is it ok for me to not transfer the autoloads folder and keep using the old one? or are there changes in there too that i need? because i have a change that i find i keep having to transfer between releases. because you defined a class called Player in the PlayerClient but as common with many games i already have a player class so i keep having to change the one in the plugin to PlayerInfo 😅 
and another thing, i notice in the cfg file version is 1.8.2 but in the release its 1.8.3. im assuming its ok for me to change it manually?

**Comment by Iakobs** [comment]
That code snipet is correct, yes!

To install the version you can only install the changed files + the bin folder. You can check what files changed in the commit, but it's basically the `snapshots_client.gd` and the `json_marshaller.gd`. About the issue with the Player, I feel so dumb! 🙈 I might release a new version fixing that, I wonder how no one told me before to change it! Thanks for letting me know!!

And one last thing, the version in the cfg file is updated now, just download the `addons.zip` file again.

**Comment by Iakobs** [comment]
Hi again! I just created a new major version renaming all `Player` related classes to `PlayGamesPlayer`. Hope that helps in the future! Check the [release note](https://github.com/Iakobs/godot-play-game-services/releases/tag/v2.0.0) for a more detailed explanation of the changes!

**Comment by TheSkyOne** [comment]
im not super well-versed with using github yet so im struggling to find how to access the commit to see the changes 😅 
tho since this is a compatibility breaking release and since you have changed the name of the Player class (which was the main thing i changed for my own, thank you for doing that 😄), i decided to delete the entire plugin and reinstall all of it.
tho now i am getting this issue
![Screenshot 2024-10-11 150034](https://github.com/user-attachments/assets/458e8eb6-a6ab-4215-a4f8-c4462943a88c)
when trying to enable the plugin in the project settings.
by the way, what is the JSON marshaller for? i dont think i need it since i use only logging-in and snapshots, but i dont want to delete it without understanding what it does.

**Comment by TheSkyOne** [comment]
yeah i have tried restarting the project. still it gives the same error when trying to enable it.

well thats good to know.
> You actually need all the files in the plugin.
does that mean that even if i dont plan to use some of the clients such as the achievements clients i should still keep the script around and just disable it in the globals tab?

**Comment by Iakobs** [comment]
Maybe you don't need all the files, but some of them have dependencies between them, that's why I wouldn't recommend deleting any files, unles you know exactly what you are doing. I'll also try not disabling anything until you're sure everything works. Then you can try disabling the autoloads that you don't need and check if everything still works fine.

About the error you mentioned, there's any other error message? The editor usually shows more messages in the output tab in the bottom dock.

**Comment by TheSkyOne** [comment]
im sorry, that was totally my mistake. i forgot i removed the marshaller even before i asked you the question and that caused the issue.
i am also seeing now that i am getting null for when there is not snapshot, so that seems to work as well!
thanks for adding this and all the support for the plugin 😄

**Comment by Iakobs** [comment]
nice! I'm glad you're progressing!! As always, I'll close the issue now but feel free to ask anything!
